src/transformStateWithClones.js

```diff
@@ -6,8 +6,31 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateHistory = [];
+  let currentState = { ...state };
```

Comment:
It's good that you are making a copy of the state object to avoid mutations. However, according to the checklist item 5, it would be better to use a more descriptive name for the copy of the state object, like `stateCopy` instead of `currentState` to improve code readability.